### ISSUE TITLE
Disallow usage of `>-`, which is not supported by GitHub actions

### ISSUE THREAD

**Issue by sarayourfriend** [rhysd/actionlint#397]
GitHub actions subset of YAML does not support `>-`: https://github.com/actions/runner/issues/418#issuecomment-612928525

A lint error should be raised when `>-` is used to prevent the headache of finding out it doesn't work.

**Comment by rhysd** [comment]
I didn't know this behavior. It looks nice if actionlint can catch it. I need to check how go-yaml parses the syntax. Can you provide some sample YAML input which should cause the error?

**Comment by rhysd** [comment]
I tried this on my test repository but it looks that `>-` is supported.

Workflow was:

```yaml
on: push

jobs:
  test:
    runs-on: ubuntu-latest
    steps:
      - run: >-
          echo foo
          bar
```

And `echo foo bar` seemed to be executed correctly.

![image](https://github.com/rhysd/actionlint/assets/823277/9ddda75d-7005-4144-ab66-207828feca57)

Am I misunderstanding anything?

**Comment by sarayourfriend** [comment]
Interesting! I had a PR that would use this in a workflow on WordPress/openverse, and another maintainer had run into an issue before with `>`, and advised me to avoid it, including the link to the GitHub actions issue I shared in this issue's description.

I'm realising now that I didn't actually get a chance to try the workflow I was modifying (it only runs when we publish a release, so not in a PR), and I did not look deeply into the issue the other maintainer had. It _seems_ like the other issue may have been unrelated to the `>` usage (it wasn't even `>-`).

Anyway, I'm playing around now with the original failure that the other maintainer ran into, and it turns out it has to do with the leading whitespace on the other lines of the block, which you've left out.

I think GitHub's YAML parser works perfectly, the Openverse project and the person who opened https://github.com/actions/runner/issues/418, just didn't understand (or expect) how YAML parses leading whitespace on lines inside a `>` block.

Compare these two: [without leading whitespace on the second line](https://yaml-online-parser.appspot.com/?yaml=run%3A+%3E%0A++++echo+foo%0A++++bar%0A&type=json) and [with leading whitespace](https://yaml-online-parser.appspot.com/?yaml=run%3A+%3E%0A++++echo+foo%0A++++++++bar%0A&type=json). I, and I bet everyone else having this issue, incorrectly assumed the whitespace would be treated just as spaces. However, the YAML spec (somewhat inexplicably, if you ask me, but :shrug:), thinks there should be a newline in those cases.

Here's the relevant section in the YAML spec: https://yaml.org/spec/1.2.2/#block-folding

> folding does not apply to [line breaks](https://yaml.org/spec/1.2.2/#line-break-characters) surrounding text lines that contain leading [white space](https://yaml.org/spec/1.2.2/#white-space-characters)

Meaning, newlines are inserted when there is leading whitespace.

So, while I _wish_ this was something actionlint could solve, I don't think it's within actionlint's scope. It's really better put to a YAML linter to warn you that this feature might have some pitfalls. I'm not confident either way, but I think you'd be fine to close this issue as out of scope.

Thanks for taking the time to look at this!

**Comment by rhysd** [comment]
I commented to the original issue.

https://github.com/actions/runner/issues/418#issuecomment-1953519221

**Comment by rhysd** [comment]
Thank you for the clarification. Then I'm closing this issue.
